### PR TITLE
PLT-7709 Update plugin settings in the system console

### PR DESCRIPTION
**Pull request by jwilander**
#### Summary
Update plugin settings in the system console.

#### Ticket Link
Part 1 of https://mattermost.atlassian.net/browse/PLT-7709

#### Checklist
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

**Comment by jwilander** [comment]
@jasonblais the styling might be slightly off in some places. For example, there's no border between multiple plugins on the management page

File: components/admin_console/plugin_management/plugin_management.jsx [hunk 1]
```diff
@@ -103,7 +115,145 @@ export default class PluginSettings extends React.Component {
...
+    }
+
+    buildPluginItem = (p) => {
+        let activateButton;
+        if (p.active) {
+            const deactivating = this.state.deactivating === p.id;
+            activateButton = (
+                <a
+                    disabled={deactivating}
+                    onClick={() => this.handleDeactivate(p.id)}
```
Comment (enahum) [inline components/admin_console/plugin_management/plugin_management.jsx]: is there a way you can remove the arrow function here and in the other places you do the same thing?

it seems that arrow functions or bindings affect performance as every time it renders it creates a new function

**Comment by jasonblais** [comment]
@jwilander

1) When plugins are disabled, the plugin upload setting shouldn't be editable

image

2) When plugins and plugin uploads are enabled, I get an error `Plugins have been disabled.` when trying to upload one:
image

3) Added `admin.plugin.installedDesc` to en.json and did two minor UI text updates based on latest feedback from LB.

Once 2 is fixed, I'll see if @asaadmahmood has time with CSS tweaks on the plugin list. It uses the same styling from Account Settings > Personal Access Tokens list, so I think it would be pretty quick.

Otherwise looks great. Okay to hide the controls on Management for now when plugins are disabled. It's still obvious what's going on, which is most important.

**Comment by jwilander** [comment]
1. Will update
2. There's an issue on the server that's unrelated to this PR causing this. It should work fine if plugins are already enabled and you don't disable them

**Comment by jasonblais** [comment]
2. Ah okay, sounds good

**Comment by jasonblais** [comment]
Is there a separate PR to fix that?

File: components/admin_console/plugin_management/plugin_management.jsx (hunk 1 above)
Comment (jwilander) [reply components/admin_console/plugin_management/plugin_management.jsx]: I could probably do it but I'm pretty sure it's only a problem for anything that's a PureComponent or using shouldComponentUpdate, which this isn't

**Comment by jwilander** [comment]
Not yet, it looks likes there's a couple of issues causing it. One related to my changes and one related to some refactoring of how the config file is handled

**Comment by jasonblais** [comment]
Sounds good, I created a ticket so we don't lose track of it. Fine to fix separately: https://mattermost.atlassian.net/browse/PLT-8036

**Comment by jwilander** [comment]
Fix for 1 pushed

**Comment by jasonblais** [comment]
https://github.com/mattermost/mattermost-server/pull/7770 should fix 2. Verified 1.

I was still having trouble uploading plugins on the newest spinmint, but maybe it's cause the build didn't pass (?). AM would need it so he can tweak the styling of the plugin list

**Comment by jwilander** [comment]
@jasonblais there was a file permissions issue on the spinmint which was preventing start-up of plugins. It should be working on that spinmint now

**Comment by asaadmahmood** [comment]
@jasonblais Done.

**Pull request merged**